cap-primitives/src/rustix/fs/mod.rs

```diff
@@ -142,10 +142,13 @@ fn tty_path() {
     #[cfg(unix)]
     use std::os::unix::fs::FileTypeExt;
 
-    // On FreeBSD, /dev/{tty,stdin,stdout,stderr} are aliases to different real
-    // devices.
     let paths: &[&str] = if cfg!(target_os = "freebsd") {
+        // On FreeBSD, /dev/{tty,stdin,stdout,stderr} are aliases to different
+        // real devices.
         &["/dev/ttyv0", "/dev/pts/0"]
+    } else if cfg!(target_os = "illumos") {
```

Comment:
I wasn't sure about Solaris here so I left this as illumos. The other bits I'm reasonably sure are okay on Solaris.